_minutes/2024-12-05-wecg.md

```diff
@@ -0,0 +1,178 @@
+ * [rob] That sounds useful. On the fly use case, If an extension wants to create a bookmark in the toolbar and make sure they're not synced, how would a developer specify that?
+ * [monica] I don't think you'd be able to do so. In general today when you create a bookmark you need to create it in a given folder. When you decide where to put a bookmark, can check whether or not the folder is syncable. Devs should not assume whether or not a folder is syncable.
+ * [james] To take a concrete example, if the user is not logged in there will be no syncable folders. That's the opposite of your example, but devs shouldn't assume all content is of a single type.
+ * [oliver] Agree with the use case of a fully private extension, but if we cannot guarantee it in the browser, we cannot implement it.
+ * [monica] Not possible to add restrictions to bookmarks to prevent an item from being moved to an account folder.
+ * [rob] Firefox is neutral on this.
+ * [timothy] Neutral as well for Safari.
+
+[Issue 539](https://github.com/w3c/webextensions/issues/539): RegisteredContentScript.tabIds and RegisteredContentScript.excludeTabIds properties to filter injection
+
+ * [giorgio] When we last discussed this issue Oliver noted that there was a proposal to inject properties. I wanted to check if there was any progress on supporting globals for the register content script API. What I've gathered porting NoScript to MV3 is that we've got a lot of bug reports related to … that can be covered by DNR session rules + sync XHR. If scripting.registerContentScripts can support similar filters (URL or tabId filter), I wouldn't have to use such fragile work-arounds.
+ * [rob] Question: you mentioned no support for register by URL. Content script matching by URL is supported, isn't it?
+ * [giorgio] Parameters. Can declare fixed scripts but cannot set parameters for content scripts by tabID or URL.
+ * [rob] If you had a choice, would you prefer a way to declare parameters per URL or have sync IPC?
+ * [giorgio] Sync IPC would be great, but I don't expect you'd want that. Just need sync messaging once at tab start.
+ * [rob] Sync IPC would be unlikely. A more reasonable proposal may be synchronously available storage available in a content script. What kind of quota would you need?
+ * [giorgio] If you can't filter by URL and tab, I'm not sure. It depends on the user.
+ * [oliver] Global params or per-tab params are definitely still the path we'd like to take. I realize there's been a multi-year delay. Realize adding per-tab is similar in scope to global params, so neither path is obviously easier than the other.
+ * [giorgio] 10 MB if there's no filter.
+ * [rob] That's … a lot. From the API design point of view, it would be preferable for extensions to be able to specify different values, so that they are not forced to use 1 giant global “scriptParams”.
```

Comment:
```suggestion
 * [rob] That's … a lot. From the API design point of view, it would be preferable for extensions to be able to specify different values, so that they are not forced to use one giant global "scriptParams".
```